keras/callbacks.py

```diff
@@ -737,10 +739,21 @@ def on_epoch_end(self, epoch, logs=None):
                     tensors += [K.learning_phase()]
 
                 assert len(val_data) == len(tensors)
-                feed_dict = dict(zip(tensors, val_data))
-                result = self.sess.run([self.merged], feed_dict=feed_dict)
-                summary_str = result[0]
-                self.writer.add_summary(summary_str, epoch)
+                val_size = val_data[0].shape[0]
+                i = 0
+                while i < val_size:
+                    step = min(self.batch_size, val_size - i)
+                    batch_val = []
+                    batch_val.append(val_data[0][i:i + step - 0])
+                    batch_val.append(val_data[1])
+                    batch_val.append(val_data[2])
```

Comment:
`val_data[1]` and `val_data[2]` are targets and sample_weights, right? So don't you need to slice them as well?